solr/core/src/java/org/apache/solr/core/NodeConfig.java

```diff
@@ -206,7 +208,16 @@ public Path getSolrDataHome() {
     return solrDataHome;
   }
 
-  /** 
+  /**
+   * Obtain the path of solr's binary installation directory, e.g. <code>/opt/solr</code>
+   * @return path to install dir, or null if property 'solr.install.dir' has not been initialized
+   */
+  public Path getSolrInstallDir() {
+    String prop = System.getProperty(SolrDispatchFilter.SOLR_INSTALL_DIR_ATTRIBUTE);
+    return prop != null ? Paths.get(prop) : null;
```

Comment:
I was a little weary about returning null here. Alternatives would be String which could be "" or `Optional<Path>` but since solr.install.dir is always set, both in a normal run and in test, that is a corner case I did not bother with.

Reply:
I also thought of not adding this method to `NodeConfig` and instead use `System.getProperty` directly where you need it - like is done other places in the code base. But it IS a node configuration and fits nicely in there. Wdyt?

Reply:
I like that you placed this here.  Furthermore, perhaps SOLR_INSTALL_DIR_ATTRIBUTE actually belongs here.

Reply:
DispatchFilter has six different sysprop constants defined already. I think perhaps instead of spreading them all around we should factor out a new class `SolrSystemProperties` to keep all sysprop contants in one place? But that is a refactoring for another day.